File: src/main/kotlin/client/ui/Button.java
package client.ui;

import client.renderables.Drawable;
import org.jetbrains.annotations.NotNull;

import java.awt.*;

public class Button implements Drawable {
    public int x, xBound, y, yBound;
    private ActionButton action;

    public Button(int x, int xBound, int y, int yBound, ActionButton action) {
        this.x = x;
        this.xBound = xBound;
        this.y = y;
        this.yBound = yBound;
        this.action = action;
    }

    public void onClick(int x, int y) {
        if ((x >= this.x && x <= (this.x+this.xBound))&&(y >= this.y && x <= (this.y+this.yBound))) {
            action.action();
        }
    }

    @Override
    public void draw(@NotNull Graphics g) {}
}

File: src/main/kotlin/client/ui/Button.kt
package client.ui

import client.renderables.Drawable
import java.awt.Graphics

open class Button(var x: Int, var xBound: Int, var y: Int, var yBound: Int, private val action: () -> Unit) : Drawable {

    fun onClick(x: Int, y: Int) {
        if (x >= this.x && x <= this.x + this.xBound && y >= this.y && x <= this.y + this.yBound) {
            action.invoke()
        }
    }

    override fun draw(g: Graphics) {}
}